test/modules/obsagent-client/src/config.rs

```diff
@@ -0,0 +1,84 @@
+use std::error::Error;
+
+use clap::{value_t, App, Arg};
+
+#[derive(Debug)]
+pub struct Config {
+    pub update_rate: f64,
+    pub prom_config: PromConfig,
+    pub otel_config: OTelConfig,
+}
+
+#[derive(Debug)]
+pub struct OTelConfig {
+    pub push_rate: f64,
+    pub otlp_endpoint: String,
+}
+
+#[derive(Debug)]
+pub struct PromConfig {
+    pub endpoint: String,
+}
+
+pub fn init_config() -> Result<Config, Box<dyn Error + Send + Sync + 'static>> {
+    let matches = App::new("obs_agent_client")
+        .arg(
+            Arg::with_name("update-rate")
+                .short("u")       
+                .long("update-rate")
+                .takes_value(true)
+                .help("Rate at which each instrument is updated with a new metric measurement (updates/sec)")
+        )
+        .arg(
+            Arg::with_name("push-rate")
+                .short("p")
+                .long("push-rate")
+                .takes_value(true)
+                .help("Rate at which measurements are pushed out of the client (pushes/sec)")
```

Comment:
From my DSP background, I feel like by having 
1. Have the `update-rate` and `push-rate` being floating points
2. Not restrict them to be the same rate

open another can of worm that we may not want to deal with. Say... A metric X `update-rate` and `push-rate` are different, do we want to deal with interpolating/decimating of the data point inbetween them? Likewise for each individual rate... do we want to be responsible for estimating the data that's not being sampled?

Reply:
So, the `update-rate` and `push-rate` actually have a slightly different meaning than what you're thinking. The `update-rate` just specifies how often the obsagent-client records a new measurement using each instrument.  In OpenTelemetry, these recorded measurements are stored by a 'push controller' and periodically pushed out to any configured exporters.  In this obsagent-client, the `push-rate` parameter is used to specify the rate at which the push controller sends collected measurements to the exporters.  Does that clarify things?

Reply:
Yup. It's the rate which we trigger the end-point. That make sense. No need to deal w/ interpolation/decimation there.